src/gsi/gsi_rfv3io_mod.f90

```diff
@@ -1240,98 +1473,89 @@ subroutine gsi_fv3ncdf_read_v1(filenamein,varname,varname2,work_sub,mype_io)
 
 
     use kinds, only: r_kind,i_kind
-    use mpimod, only: ierror,mpi_comm_world,npe,mpi_rtype,mype
-    use gridmod, only: lat2,lon2,nsig,nlat,nlon,itotsub,ijn_s
+    use mpimod, only:  mpi_rtype,mpi_comm_world,mype,MPI_INFO_NULL
+    use mpimod, only: mpi_comm_world,mpi_rtype,mype
     use netcdf, only: nf90_open,nf90_close,nf90_get_var,nf90_noerr
     use netcdf, only: nf90_nowrite,nf90_inquire,nf90_inquire_dimension
     use netcdf, only: nf90_inquire_variable
     use netcdf, only: nf90_inq_varid
     use mod_fv3_lola, only: fv3_h_to_ll
-    use general_commvars_mod, only: ltosi_s,ltosj_s
+    use gsi_bundlemod, only: gsi_bundle
+    use general_sub2grid_mod, only: sub2grid_info,general_grid2sub
 
     implicit none
-    character(*)   ,intent(in   ) :: varname,varname2,filenamein
-    real(r_kind)   ,intent(out  ) :: work_sub(lat2,lon2,nsig) 
-    integer(i_kind)   ,intent(in   ) :: mype_io
-    character(len=128) :: name
-    real(r_kind),allocatable,dimension(:,:,:):: uu
-    real(r_kind),allocatable,dimension(:,:,:):: temp0 
-    integer(i_kind),allocatable,dimension(:):: dim
-    real(r_kind),allocatable,dimension(:):: work
-    real(r_kind),allocatable,dimension(:,:):: a
-
-
-    integer(i_kind) n,ns,k,len,var_id
+    type(sub2grid_info), intent(in):: grd_ionouv 
+    character(len=max_varname_length),intent(in):: filenamein
```

Comment:
For an input parameter, it should be fine to say "character(*)   ,intent(in)::filenamein 
No need to use "max_varname_length" here and in other similar places.

Reply:
Changed as suggested.